spec/01-unit/26-mime-type_spec.lua

```diff
@@ -0,0 +1,94 @@
+          mime_type = "application/json; charset=UTF-8",
+          result = { type = "application", subtype = "json", params = { charset = "UTF-8" } }
+        },
+        {
+          -- multiple parameters
+          mime_type = "application/json; charset=UTF-8; Key=Value; q=1",
+          result = { type = "application", subtype = "json", params = { charset = "UTF-8", Key = "Value", q = "1" } }
+        },
+        {
+          -- malformed whitespace
+          mime_type = "application/json ;  charset=UTF-8 ; Key=Value",
+          result = { type = "application", subtype = "json", params = { charset = "UTF-8", Key = "Value" } }
+        },
+        {
+          -- quote parameter value
+          mime_type = 'application/json; charset="UTF-8"',
+          result = { type = "application", subtype = "json", params = { charset = "UTF-8" } }
+        },
+        {
+          -- opts.param_key_ignorecase
+          mime_type = "application/json; Charset=UTF-8; Key=Value",
+          opts = { param_key_ignorecase = true },
+          result = { type = "application", subtype = "json", params = { charset = "UTF-8", key = "Value" } }
+        },
+        {
+          -- opts.ignorecase_params
+          mime_type = "application/json; Charset=UTF-8; Key=Value",
+          opts = { ignorecase_params = { "charset" } },
+          result = { type = "application", subtype = "json", params = { charset = "UTF-8", Key = "Value" } }
+        },
+        {
+          -- opts.param_key_ignorecase and opts.ignorecase_params, param_key_ignorecase will take effect
+          mime_type = "application/json; Charset=UTF-8; Key=Value",
+          opts = { param_key_ignorecase = true, ignorecase_params = { "charset" } },
+          result = { type = "application", subtype = "json", params = { charset = "UTF-8", key = "Value" } }
+        },
+        {
+          mime_type = "*/*; charset=UTF-8; q=0.1",
+          result = { type = "*", subtype = "*", params = { charset = "UTF-8", q = "0.1" } }
+        },
+        {
+          mime_type = "application/*",
+          result = { type = "application", subtype = "*", params = {} }
+        },
+        {
+          mime_type = "*/text",
+          result = { type = "*", subtype = "text", params = {} }
+        },
+        {
+          mime_type = "*",
+          result = { type = "*", subtype = "*", params = {} }
+        },
+        {
+          mime_type = "*; q=.2",
+          result = { type = "*", subtype = "*", params = { q = '.2' } }
+        },
+        {
+          -- invalid input
+          mime_type = "helloworld",
+          result = { type = nil, subtype = nil, params = nil }
+        },
+        {
+          -- invalid input
+          mime_type = " ",
+          result = { type = nil, subtype = nil, params = nil }
+        },
+        {
+          -- invalid input
+          mime_type = "application/json;",
+          result = { type = nil, subtype = nil, params = nil }
+        }
```

Comment:
can we add a test for when the mime type is invalid?

Reply:
Do you mean move the invalid test cases at another `it`?

Reply:
no sorry I meant add a test case where the mime type is invalid ... if there is one I missed it my bad